packages/jest-types/src/Global.ts

```diff
@@ -123,5 +123,5 @@ export interface GlobalAdditions extends TestFrameworkGlobals {
 export interface Global
   extends GlobalAdditions,
     Omit<typeof globalThis, keyof GlobalAdditions> {
-  [extras: string]: unknown;
+  [extras: string | symbol]: unknown;
```

Comment:
Hm.. Perhaps instead of tweaking this one, it would be better to extend `Global` locally inside [jest-runtime/src/types.ts](https://github.com/facebook/jest/blob/main/packages/jest-runtime/src/types.ts). In somewhat similar way:

https://github.com/facebook/jest/blob/fe5f37038c3771aa9014bc351c0bddb04c35da7d/packages/jest-circus/src/types.ts#L10-L23